redis/tests/support/mod.rs

```diff
@@ -98,18 +102,80 @@ impl RedisServer {
                 redis::ConnectionAddr::Unix(PathBuf::from(&path))
             }
         };
-        RedisServer::new_with_addr(addr, None, |cmd| {
-            cmd.spawn()
-                .unwrap_or_else(|err| panic!("Failed to run {:?}: {}", cmd, err))
-        })
+        RedisServer::new_with_addr(
+            addr,
+            None,
+            |cmd| {
+                cmd.spawn()
+                    .unwrap_or_else(|err| panic!("Failed to run {:?}: {}", cmd, err))
+            },
+            None,
+        )
+    }
+
+    pub fn new_with_modules(modules: Vec<RedisModules>) -> RedisServer {
```

Comment:
This can take a `&[RedisModules]` instead, since we presumably don't need to mutate this? Also I think naming this `with_modules()` (omitting the `new_` prefix) is more idiomatic.

Also we don't want to duplicate `new()` here, `new()` should just call `with_modules()` with an empty `modules` argument.